src/openrct2/interface/viewport_interaction.c

```diff
@@ -220,15 +220,15 @@ sint32 viewport_interaction_get_item_right(sint32 x, sint32 y, viewport_interact
         if (tile_element_get_type(tileElement) == TILE_ELEMENT_TYPE_PATH)
             return info->type = VIEWPORT_INTERACTION_ITEM_NONE;
 
-        ride = get_ride(tileElement->properties.track.ride_index);
+        ride = get_ride(track_element_get_ride_index(tileElement));
```

Comment:
Hmm technically this is not a track element at this point. It is either a track or a entrance element. This may cause issues if we change where the ride index is in the element structure.

Reply:
Would rewriting it as follows help:
```
if (tile_element_get_type(tileElement) == TILE_ELEMENT_TYPE_TRACK)
{
    ride = get_ride(track_element_get_ride_index(tileElement));
}
else
{
    ride = get_ride(entrance_element_get_ride_index(tileElement));
}
```

Reply:
Yeah that would be better. Well safer.

Reply:
Yes, and this won't be the only case. All the elements are carefully laid out so that common properties such as ride index are in the same location. I think it would be better now is if we had a `map_element_get_ride_index` which will do the above code or error if not a valid element type that can have ride index.

Reply:
I'll make a generic function like IntelOrca described.